Best practices/Debugging Source Generator.md

```diff
@@ -0,0 +1,95 @@
+
+Fortunately, Visual Studio comes to rescue with its possibility to debug source generators after some easy steps:
+
+1. Install .NET Compiler Platform SDK 
+   
+    In Visual Studio Installer choose Modify on Visual Studio version which has source generator project and install Visual Studio extension development under tab Workloads > Other Toolsets. 
+
+    | ![visualStudioInstallerHome](/resources/visual-studio-installer-home.png) |
+    |:--:|
+    | Figure 2 - Visual Studio Installer |
+
+    | ![visualStudioModify](/resources/visual-studio-modify.png) |
+    |:--:|
+    | Figure 3 - Visual Studio modifying window |
+
+2. Add IsRoslynComponent property in source generator project file:
+   
+    ```xml
+    <Project Sdk="Microsoft.NET.Sdk">
+        <PropertyGroup>
+            <TargetFramework>netstandard2.0</TargetFramework>
+            <IsRoslynComponent>true</IsRoslynComponent>
+            <Nullable>enable</Nullable>
+            <LangVersion>latest</LangVersion>
+        </PropertyGroup>
+        <ItemGroup>
+            <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.0.1" PrivateAssets="all" />
+            <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.3">
+                <PrivateAssets>all</PrivateAssets>
+                <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+            </PackageReference>
+        </ItemGroup>
+    </Project>
+    ```
+
+3. Add reference to source generator in a project that will be using it (if it already isn't added):
+   
+    ```xml
+    <ItemGroup>
+        <ProjectReference Include="..\JsonApi.SourceGenerator\JsonApi.SourceGenerator.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="true" />
+    </ItemGroup>
+    ```
+
+4. Add debug configuration using following steps:
+   
+   1. Right click on the project that contains source generator
+   2. Click `Properties`
+   3. Click `Debug`
+   4. Click `Open debug launch profiles UI`
+      | ![debugProperties](/resources/debug-properties.png) |
+      |:--:|
+      | Figure 4 - Debug properties window |
+   5. Delete existing profiles
+      | ![deleteProfile](/resources/delete-existing-profile.png) |
+      |:--:|
+      | Figure 5 - Delete existing profile |
+   6. Add new `Roslyn Component` profile
+      | ![createProfile](/resources/create-roslyn-component-profile.png) |
+      |:--:|
+      | Figure 6 - Add new profile |
+   7. Choose target project from list of projects
+      | ![chooseTargetProject](/resources/choose-target-project.png) |
+      |:--:|
+      | Figure 7 - Choose target project |
+   8. **Optional**: Rename the debug profile so you can easily find and use it later
+
+| ![endResult](/resources/end-result.png) |
+|:--:|
+| Figure 7 - Debug profile after creation |
+
+After these steps you will need to restart Visual Studio and set your source generator project as a startup project. 
```

Comment:
I think this could be added as the last step of the process since the whole setup won't work without it. I know I would have missed it in this form if I wasn't reading this as a reviewer.